src/exeploreGame/GamePannel4.java

```diff
@@ -61,11 +60,11 @@ private void myinit() {
         controlPannel.setFocusable(true);
         controlPannel.setOpaque(false);//to make buttons transparent
         controlPannel.setFocusTraversalKeysEnabled(false);
-        keyInputAdapter=new KeyAdapter(){
-            public void keyTyped(KeyEvent e){
+        final KeyAdapter keyInputAdapter = new KeyAdapter() {
+            public void keyTyped(KeyEvent e) {
                 keyInput(e);
             }
-        };//calls keyInput()
```

Comment:
these comments are for understandability .
so that reader can judge what is happening